test/integration/model/attributes/types.test.js

```diff
@@ -144,6 +154,54 @@ describe(Support.getTestDialectTeaser('Model'), function() {
             expect(users[0].storage).to.equal('something');
           });
         });
+
+        it('should be populated with default values when loaded from the db', function () {
+          var self = this;
+          return this.User.create({
+            field1: 'fizzbuzz',
+          }).then(function() {
+            return self.User.findOne({ where: { storage: 'fizzbuzz' }});
+          }).then(function (user) {
+            expect(user.virtualWithDefault).to.equal('cake');
+            expect(user.virtualWithoutDefault).to.equal(undefined);
+          });
+        });
+
+        it('should be populated to a default value on reload() if not previously set', function () {
+          return this.User.create({
+            field1: 'fizzbuzz',
+          }).then(function(user) {
+            expect(user.virtualWithDefault).to.equal('cake');
+            return user.reload();
+          }).then(function(user) {
+            expect(user.virtualWithDefault).to.equal('cake');
+            expect(user.virtualWithoutDefault).to.equal(undefined);
+            user.set('virtualWithDefault', undefined);
```

Comment:
Looking at this now, I also want to add a test case where the virtual field is not in the `attributes` option passed into reload & make sure it doesn't get set to the default value